NuRadioMC/simulation/simulation.py

```diff
@@ -1,85 +1,1032 @@
+                time_logger.stop_time('distance cut')
+                continue
+            # logger.warning(f"shower {shower.get_id()} is close enough ({np.linalg.norm(x1 - x2)/units.km:.2f} < {distance_cut(shower_energy_sum)/units.km:.2f}) to station {sid} channel {cid}, continuing with shower")
+            time_logger.stop_time('distance cut')
+        time_logger.start_time('ray tracing')
+        logger.debug(f"Calculating electric field for shower {shower.get_id()} and station {sid}, channel {cid}")
+        shower_axis = -1 * shower.get_axis() # We need the propagation direction here, so we multiply the shower axis with '-1'
+        n_index = medium.get_index_of_refraction(x1)
+        cherenkov_angle = np.arccos(1. / n_index)
+
+        p.set_start_and_end_point(x1, x2)
+        p.use_optional_function('set_shower_axis', shower_axis)
+        if config['speedup']['redo_raytracing']:  # check if raytracing was already performed
+            pass
+            # TODO: initiatlize ray tracer with existing results if available
+        p.find_solutions()
+        if not p.has_solution():
+            logger.debug(f"shower {shower.get_id()} and station {sid}, channel {cid} from {x1} to {x2} does not have any ray tracing solution")
+            continue
+
+        n = p.get_number_of_solutions()
+        logger.debug(f"found {n} solutions for shower {shower.get_id()} and station {sid}, channel {cid} from {x1} to {x2}")
+        delta_Cs = np.zeros(n)
+        viewing_angles = np.zeros(n)
+        # loop through all ray tracing solution
+        for iS in range(p.get_number_of_solutions()):
+            viewing_angles[iS] = hp.get_angle(shower_axis, p.get_launch_vector(iS))
+            delta_Cs[iS] = viewing_angles[iS] - cherenkov_angle
+        # discard event if delta_C (angle off cherenkov cone) is too large
+        if min(np.abs(delta_Cs)) > config['speedup']['delta_C_cut']:
+            logger.debug(f'delta_C too large, event unlikely to be observed, (min(Delta_C) = {min(np.abs(delta_Cs))/units.deg:.1f}deg), skipping event')
+            continue
+        time_logger.stop_time('ray tracing')
+        for iS in range(n): # loop through all ray tracing solution
+            time_logger.start_time('ray tracing (time)')
+            # skip individual channels where the viewing angle difference is too large
+            # discard event if delta_C (angle off cherenkov cone) is too large
+            if np.abs(delta_Cs[iS]) > config['speedup']['delta_C_cut']:
+                logger.debug('delta_C too large, ray tracing solution unlikely to be observed, skipping ray tracing solution')
+                continue
+            # TODO: Fill with previous values if RT was already performed
+            R = p.get_path_length(iS)  # calculate path length
+            T = p.get_travel_time(iS)  # calculate travel time
```

Comment:
no capital letters for variables. no single letter variables

Reply:
This is an remnant from the old code but I really thing that these single letter variables hurt here (also the properator `p`). They are do not serve to make the code more readable.